sympy/functions/special/delta_functions.py

```diff
@@ -236,8 +263,25 @@ def eval(cls, arg):
             return S.One
 
     def _eval_rewrite_as_Piecewise(self, arg):
-        if arg.is_real:
-            return Piecewise((1, arg > 0), (S(1)/2, Eq(arg, 0)), (0, True))
+        """Represents Heaviside in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import Heaviside, Piecewise, Symbol
+           >>> x = Symbol('x')
+
+           >>> Heaviside(x).rewrite(Piecewise)
+           Piecewise((1, x > 0), (1/2, Eq(x, 0)), (0, True))
+
+           >>> Heaviside(x - 5).rewrite(Piecewise)
+           Piecewise((1, x - 5 > 0), (1/2, Eq(x - 5, 0)), (0, True))
+
+           >>> Heaviside(x**2 - 1).rewrite(Piecewise)
+           Piecewise((1, x**2 - 1 > 0), (1/2, Eq(x**2 - 1, 0)), (0, True))
+
+        """
+        return Piecewise((1, arg > 0), (S(1)/2, Eq(arg, 0)), (0, True))
```

Comment:
But it is not `nan` right. It is either S(1/2) or S(1). I think it would be more appropriate if it outputs nothing.  
```py
In [2]: Heaviside(0)
Out[2]: Heaviside(0)

In [6]: Heaviside(x - 4).rewrite(Piecewise)
Out[6]: Piecewise((1, x-4 > 0), (0, True))
```